packages/dd-trace/src/encode/notepack.js

```diff
@@ -0,0 +1,346 @@
+  return size;
+}
+
+function _encode(bytes, defers, value) {
+  let hi = 0, lo = 0, length = 0, size = 0;
+
+  switch (typeof value) {
+    case 'string':
+      if (value.length > MICRO_OPT_LEN) {
+        length = Buffer.byteLength(value);
+      } else {
+        length = utf8Length(value);
+      }
+
+      if (length < 0x20) { // fixstr
+        bytes.push(length | 0xa0);
+        size = 1;
+      } else if (length < 0x100) { // str 8
+        bytes.push(0xd9, length);
+        size = 2;
+      } else if (length < 0x10000) { // str 16
+        bytes.push(0xda, length >> 8, length);
+        size = 3;
+      } else if (length < 0x100000000) { // str 32
+        bytes.push(0xdb, length >> 24, length >> 16, length >> 8, length);
+        size = 5;
+      } else {
+        throw new Error('String too long');
+      }
+      defers.push({ str: value, length: length, offset: bytes.length });
+      return size + length;
+    case 'number':
+      // TODO: encode to float 32?
+
+      if (Math.floor(value) !== value || !isFinite(value)) { // float 64
+        bytes.push(0xcb);
+        defers.push({ float: value, length: 8, offset: bytes.length });
+        return 9;
+      }
+
+      if (value >= 0) {
+        if (value < 0x80) { // positive fixnum
+          bytes.push(value);
+          return 1;
+        }
+
+        if (value < 0x100) { // uint 8
+          bytes.push(0xcc, value);
+          return 2;
+        }
+
+        if (value < 0x10000) { // uint 16
+          bytes.push(0xcd, value >> 8, value);
+          return 3;
+        }
+
+        if (value < 0x100000000) { // uint 32
+          bytes.push(0xce, value >> 24, value >> 16, value >> 8, value);
+          return 5;
+        }
+        // uint 64
+        hi = (value / Math.pow(2, 32)) >> 0;
+        lo = value >>> 0;
+        bytes.push(0xcf, hi >> 24, hi >> 16, hi >> 8, hi, lo >> 24, lo >> 16, lo >> 8, lo);
+        return 9;
+      } else {
+
+        if (value >= -0x20) { // negative fixnum
+          bytes.push(value);
+          return 1;
+        }
+
+        if (value >= -0x80) { // int 8
+          bytes.push(0xd0, value);
+          return 2;
+        }
+
+        if (value >= -0x8000) { // int 16
+          bytes.push(0xd1, value >> 8, value);
+          return 3;
+        }
+
+        if (value >= -0x80000000) { // int 32
+          bytes.push(0xd2, value >> 24, value >> 16, value >> 8, value);
+          return 5;
+        }
+        // int 64
+        hi = Math.floor(value / Math.pow(2, 32));
+        lo = value >>> 0;
+        bytes.push(0xd3, hi >> 24, hi >> 16, hi >> 8, hi, lo >> 24, lo >> 16, lo >> 8, lo);
+        return 9;
+      }
+      break;
+    case 'object':
+      // nil
+      if (value === null) {
+        bytes.push(0xc0);
+        return 1;
+      }
+
+      // uint 64
+      if (value._isUint64BE) {
+        value = value.toArray();
+        bytes.push(0xcf, value[0], value[1], value[2], value[3], value[4], value[5], value[6], value[7]);
```

Comment:
Can you not just destructure/splat the `value` array here?

Reply:
I was trying to stay in line with the rest of the library which doesn't seem to use this operator. In any case, there is no performance difference between the 2.